### ISSUE TITLE
Apache 2.4 Support

### ISSUE THREAD

**Issue by diemuzi** [ByteInternet/libapache-mod-fastcgi#1]
Fails to compile with Apache 2.4

**Comment by allardhoeve** [comment]
Hey Diemuzi,

It compiled for me with Apache 2.4. Do you have a build log?

Make sure you use the Byte branch. Also make sure you apply the patches that are kept in debian/patches/.

To apply the patch that enables compiling against Apache 2.4, try:

``` bash
patch -p1 < debian/patches/byte-compile-against-apache24.diff
```

Thanks!

Allard

**Comment by diemuzi** [comment]
@allardhoeve Thanks for the update and I've tried to make this work two ways, I've created some pastebins for your review...

#

# Manaul build using the make command

#

http://pastebin.com/KWeh0DFw

#

# APXS build

#

http://pastebin.com/kE1vw3xf

In both pastes you'll see my exact commands along with the exact errors produced.

I cloned your GIT repository as of March 8 @ 8:30am just to be certain I had all of your updates. Here is my setup:

Archlinux 32 and 64 bit
Apache 2.4.1

I'd love to see this work and would enjoy helping with anything possible to make it work =)
~ Sammie

**Comment by allardhoeve** [comment]
Hey there. You fail to apply the patch for some reason. This works for me fine:

allard@allard:~/devel$ git clone git@github.com:ByteInternet/libapache-mod-fastcgi.git
Cloning into libapache-mod-fastcgi...
remote: Counting objects: 279, done.
remote: Compressing objects: 100% (145/145), done.
remote: Total 279 (delta 159), reused 246 (delta 126)
Receiving objects: 100% (279/279), 159.41 KiB, done.
Resolving deltas: 100% (159/159), done.

allard@allard:~/devel$ cd libapache-mod-fastcgi/

allard@allard:~/devel/libapache-mod-fastcgi$ patch -p1 < debian/patches/
byte-compile-against-apache24.diff         byte-new-packet-type-byte_acc.diff         series  
byte-fix-cast-warning-in-fcgi_config.diff  doc-misc-typo-fix.diff  
allard@allard:~/devel/libapache-mod-fastcgi$ patch -p1 < debian/patches/byte-compile-against-apache24.diff 
patching file fcgi.h
Hunk #3 succeeded at 364 (offset -11 lines).
patching file mod_fastcgi.c
Hunk #2 succeeded at 3013 (offset -5 lines).
allard@allard:~/devel/libapache-mod-fastcgi$

**Comment by diemuzi** [comment]
@allardhoeve Thanks for the quick update. I got it working but it's weird how...
- Cloning the GIT repository resulted in the patches failing.
- Downloading the ZIP version of the GIT repository resulted in the patches working just fine.

I've been researching ever since the day Apache 2.4 was released for a proper solution and thankfully someone like you was around to help make that happen. Thank you!
